core/client/components/character-name-text.js

```diff
@@ -64,8 +56,43 @@ class CharacterNameText extends Phaser.GameObjects.Container {
   setText(name, baseline) {
     this.name.setText(name);
     this.setBaseline(baseline);
+    this.updateIconPosition();
+    return this;
+  }
+
+  updateIconPosition() {
+    if (this.icon) this.icon.setX(-(this.name.width + this.icon.displayWidth) / 2 - iconMargin);
+  }
+
+  setIcon(icon) {
+    this.destroyIcon();
+    if (icon) {
+      game.scene.getScene('BootScene').loadImagesAtRuntime([icon], () => {
```

Comment:
@Donorhan Not sure about this but I think loading all guilds icons at preload is not a good solution either (can be better but depending on the number of existing guilds).
I thought about using this solution (loading at runtime) and optimize it a bit with a local cache but it seems that the LoaderPlugin already has a cache system.
Do you have a suggestion on what would be the optimal solution in your idea?

Reply:
@Donorhan I added an example of a guild with an icon to insert in DB if you want to test more easily this PR ;)

Reply:
Thank you! As soon as I find some time I'll take care of it 😬

Reply:
FWIW this is the most blocking of our PRs, we have a few branches that depend on it (such as #141 but also others that we didn't PR yet). 😅